controllers/userController.js

```diff
@@ -1,28 +1,248 @@
 const helpers = require('../_helpers')
 
+const bcrypt = require('bcryptjs')
+const imgur = require('imgur')
+const IMGUR_CLIENT_ID = process.env.IMGUR_CLIENT_ID
+
+const db = require('../models')
+const User = db.User
+const Tweet = db.Tweet
+const Reply = db.Reply
+const Like = db.Like
+
 const userController = {
-  readUser: (req, res) => {
-    console.log(helpers.getUser(req).toJSON())
-    res.send(`Hello User ${helpers.getUser(req).name}!`)
+  readUser: (req, res, next) => {
+    const id = Number(req.params.id)
+    User.findOne({
+      where: { id },
+      include: [
+        Tweet,
+        { model: User, as: 'Followers' },
+        { model: User, as: 'Followings' },
+      ]
+    }).then(user => {
+      if (!user) return res.status(404).json({
+        message: `this user(id: ${id}) do not exist!`
+      })
+
+      const userObj = Object.fromEntries(Object.entries(user.toJSON()).slice(0, 11))
+      userObj.isSelf = (user.id === req.user.id)
+      userObj.isFollowed = user.Followers.map(follower => follower.id).includes(helpers.getUser(req).id)
+      userObj.tweetsCount = user.Tweets.length
+      userObj.followingsCount = user.Followings.length
+      userObj.followersCount = user.Followers.length
+      return res.json(userObj)
```

Comment:
```suggestion
      const userObj = Object.fromEntries(Object.entries(user.toJSON()).slice(0, 11))
      userObj.isSelf = (user.id === req.user.id)
      userObj.isFollowed = user.Followers.map(follower => follower.id).includes(helpers.getUser(req).id)
      userObj.tweetsCount = user.Tweets.length
      userObj.followingsCount = user.Followings.length
      userObj.followersCount = user.Followers.length
      return res.json(userObj)
```
這段的格式要不要與其他的格式一樣
例如:
 tweets = tweets.map(tweet => ({
        ...(Object.fromEntries(Object.entries(tweet.dataValues).slice(0, 5))),
        repliesCount: tweet.dataValues.Replies.length,
        likesCount: tweet.dataValues.Likes.length,
        isLike: tweet.dataValues.Likes.map(like => like.UserId).includes(helpers.getUser(req).id)
      }))

另外，req.user.id 應該寫成 helpers.getUser(req).id

Reply:
但是這裡的 `userObj` 是一個 `Object` 不是 `Array`，所以不需要使用到 `.map()` 來展開～

> 另外，req.user.id 應該寫成 helpers.getUser(req).id

這邊其實不影響到 測試檔，因為不會驗證這段，但是照整個檔案的用法，確實採用 `helpers.getUser(req).id` 會比較一致！感謝🙏

Reply:
我的意思是，可以像map格式一樣用冒號，而不是用等號，因為像我的readTweet雖然不用展開但也是用冒號。